From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mallikarjun Kasoju <mkasoju@nvidia.com>
Date: Mon, 7 Apr 2025 06:05:42 +0000
Subject: [PATCH 179/240] NVIDIA: SAUCE: efuse: update ecid changes

ECID is moved to separate nvmem sysfs.

Update ECID implementation to use correct (ECID) sysfs.

Bug 5214421

Change-Id: I2ca77ec91bc4aad23faa65e03113240035c82e9d
Signed-off-by: Mallikarjun Kasoju <mkasoju@nvidia.com>
(cherry picked from commit 3d0685b1f2ad42afc9d5ab8fc030f8a00ebac2ae)
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3469614
Tested-by: mobile promotions <svcmobile_promotions@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Bitan Biswas <bbiswas@nvidia.com>
Reviewed-by: mobile promotions <svcmobile_promotions@nvidia.com>
(cherry picked from commit 3f7dda17b2984f7fa36680e1e06f2eb3dc27bccd)

diff --git a/drivers/nvmem/tegra-efuse.c b/drivers/nvmem/tegra-efuse.c
index 4e857255419d3..0e2893b2bf7d6 100644
--- a/drivers/nvmem/tegra-efuse.c
+++ b/drivers/nvmem/tegra-efuse.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-only
 /*
- * Copyright (c) 2013-2023, NVIDIA CORPORATION.  All rights reserved.
+ * Copyright (c) 2013-2025, NVIDIA CORPORATION.  All rights reserved.
  */
 
 #include <linux/device.h>
@@ -404,11 +404,16 @@ static const struct tegra_efuse_soc tegra264_efuse_soc = {
 	.num_lookups = ARRAY_SIZE(tegra264_efuse_lookups),
 	.keepouts = tegra264_efuse_keepouts,
 	.num_keepouts = ARRAY_SIZE(tegra264_efuse_keepouts),
-	.size = 0x1f094,
+	.size = 0x1f054,
+};
+
+static const struct tegra_efuse_soc tegra264_efuse_ecid_soc = {
+	.size = 0x40,
 };
 
 static const struct of_device_id tegra_efuse_of_match[] = {
 	{ .compatible = "nvidia,tegra264-efuse", .data = &tegra264_efuse_soc },
+	{ .compatible = "nvidia,tegra264-efuse-ecid", .data = &tegra264_efuse_ecid_soc },
 	{ /* Sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, tegra_efuse_of_match);
-- 
2.43.0

